cogeo_mosaic/backends/s3.py

```diff
@@ -24,69 +28,71 @@ def __init__(
         mosaic_def: Optional[Union[MosaicJSON, Dict]] = None,
         client: Optional[boto3_session.client] = None,
     ):
+        """Initialize S3Backend."""
         self.client = client or boto3_session().client("s3")
         self.key = key
         self.bucket = bucket
 
         if mosaic_def is not None:
             self.mosaic_def = MosaicJSON(**dict(mosaic_def))
         else:
-            self.mosaic_def = self.read(bucket, key)
+            self.mosaic_def = self.read()
```

Comment:
I just tend to pass the non-`self` methods inside `__init__` in case in the future the objects defined in `__init__` get moved around or something